lib/src/facts/freebsd/dmi_resolver.cc

```diff
@@ -0,0 +1,36 @@
+#include <internal/facts/freebsd/dmi_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+
+#include <kenv.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace freebsd {
+
+    dmi_resolver::data dmi_resolver::collect_data(collection& facts)
+    {
+        data result;
+        result.bios_vendor = kenv_lookup("smbios.bios.vendor");
+        result.bios_version = kenv_lookup("smbios.bios.version");
+        result.bios_release_date = kenv_lookup("smbios.bios.reldate");
+        result.uuid = kenv_lookup("smbios.system.uuid");
+        result.serial_number = kenv_lookup("smbios.planar.serial");
+        result.product_name = kenv_lookup("smbios.system.product");
+        result.manufacturer = kenv_lookup("smbios.system.maker");
+
+        return result;
+    }
+
+    string dmi_resolver::kenv_lookup(const char* file)
+    {
+        char buffer[100] = {};
```

Comment:
Honestly, just was copy and pasting from existing code, if there's a better way of doing it I can fix! 👍 

https://github.com/vincentbernat/lldpd/blob/8bb212d9c5970f509d53733092c8826de5fc0862/src/daemon/dmi-freebsd.c#L36